sdk/textanalytics/ai-text-analytics/src/textAnalyticsResult.ts

```diff
@@ -128,3 +166,58 @@ export function makeTextAnalyticsErrorResult(
     error: intoTextAnalyticsError(error)
   };
 }
+
+/**
+ * combines successful and erroneous results into a single array of results and
+ * sort them so that the IDs order match that of the input documents array.
+ * @param input the array of documents sent to the service for processing.
+ * @param response the response received from the service.
+ */
+export function combineSuccessfulErroneousDocuments<T1 extends TextAnalyticsSuccessResult>(
+  input: TextDocumentInput[],
+  response: TextAnalyticsResponse<T1>
+): (T1 | TextAnalyticsErrorResult)[] {
+  return processAndcombineSuccessfulErroneousDocuments(input, response, (x) => x);
```

Comment:
I really don't understand the `(x) => x` here, I'm guessing the map function is supposed to help convert specific success results (e.g. detect language) into an analyze success result with a discriminant, but that's only used in the below full version... the distinction between `combine` and `processAndCombine` is a big vague to me

Reply:
This is to say we want to keep the incoming successful results as is. Notice that the incoming response is already typed at `T1` that has the discriminant.